Recurse handle_invalid_pipeline

rexissimus opened the pull request:
Replaces https://github.com/VisTrails/VisTrails/tree/invalid-pipeline-hides-package-errors_/loops and #901.

Before,  handle_invalid_pipelines was called up to 2 times to resolve pipeline errors. But some cases will require 3 calls, as shown by 21708fabf50470467d5e0af2017ee3128b7e861e.

This pr makes handle_invalid_pipeline call iteself when needed so that it only needs to be called once.

This is based on, and includes,  #1164.

remram44 commented on vistrails/core/vistrail/controller.py:
```diff
@@ -3011,8 +3011,10 @@ def rev_map(v):
-                        child not in tm):
+                elif (not self.show_upgrades and
+                      (child in upgrades or
+                       am[child].description == 'Upgrade') and
```
This is probably safe since only our upgrade code will ever perform upgrades.

rexissimus commented:
The test now correctly loops 5 times. There was a bug in the upgrade code (0124df5a72d31e303960db67adefb980afc15254) that replaced the descriptor for  B version 0.1 with the latest version, causing the upgrade to C to never happen.

Added test for infinite loops. 

Added debug messages and conf option from chain-upgrades-loop.

Fixed invalid pipelines causing version=Null (eff5d13dab7362bca8dc57b235ad913da8709ebf)
